kvscheduler/plugin_scheduler.go

```diff
@@ -0,0 +1,296 @@
+package kvscheduler
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	. "github.com/ligato/cn-infra/kvscheduler/api"
+
+	"github.com/ligato/cn-infra/idxmap"
+	"github.com/ligato/cn-infra/idxmap/mem"
+	"github.com/ligato/cn-infra/kvscheduler/graph"
+	"github.com/ligato/cn-infra/kvscheduler/registry"
+	"github.com/ligato/cn-infra/logging"
+)
+
+const (
+	// DependencyRelation identifies dependency relation for the graph.
+	DependencyRelation = "depends-on"
+
+	// DerivesRelation identifies relation of value derivation for the graph.
+	DerivesRelation = "derives"
+)
+
+var schedulerSingleton *Scheduler
```

Comment:
Why use singleton? Seems like this should be generic plugin like others and be as dependency for others.